### PR TITLE
Bringing openstack commts

### PR DESCRIPTION
**Pull request by ldeataid**
The purpose of this pull request is to bring the following commits from the openstack branch to the main branch:

-  [ Set requirements versions](https://github.com/Danmcaires/poc_GenAI/commit/fb23051724664effe97465ec345f5932efcf8f4c)
-  [ Add /env directory to .gitignore ](https://github.com/Danmcaires/poc_GenAI/commit/0b1157231f85d15f6082aa035f061c66efa95488)
-  [Update README info to Openstack ](https://github.com/Danmcaires/poc_GenAI/commit/196c2fd140af555050db172b595f6dcb9d24476b) (Adapted for StarlingX)
-  [Update requirements.txt path in Dockerfile](https://github.com/Danmcaires/poc_GenAI/commit/2163a6e98c050017978a3da2868f59f83c2b3260)
-  [Empty init files are required to avoid possible bugs](https://github.com/Danmcaires/poc_GenAI/commit/f82510aa55e85304986f9131a7d707e9524dd6a5)
-  [Ignoring the README.md in Docker](https://github.com/Danmcaires/poc_GenAI/commit/3cd11b059b8d2212388f1723948552ea5c1acc82)
-  [Keeping the good practice of having a main.py](https://github.com/Danmcaires/poc_GenAI/commit/523d7ad89dcc56dbca7a115f36dede97e7867ff5)
-  [Remove unnecessary command at CMD](https://github.com/Danmcaires/poc_GenAI/commit/cce76ecae1d72c9c65367c1b12fe5ef8f13b6f4c)
-  [Shrink Docker image size from 1.51GB to 527MB](https://github.com/Danmcaires/poc_GenAI/commit/71aa0ed51cea6590dbe2f7d663bc209491e528fa)
-  [Create Helm charts for app](https://github.com/Danmcaires/poc_GenAI/commit/370b16fda2982d8aaba0087c1ed719dcb237e666)  (Adapted for StarlingX)
-  [Loading openstack_request class once per session](https://github.com/Danmcaires/poc_GenAI/commit/0704d0f7b8e841e541d83bb1e68c0961dde3d5f0)  (Adapted for StarlingX)

### CODE REVIEW

File: src/api_request.py [hunk 1]
```diff
@@ -125,28 +122,13 @@ def get_API_response(self):
...
-        self.type = instance['type']
-
-        # Necessary API address
-        pattern = r"(https?)://(?:\d{1,3}\.){3}\d{1,3}:"
-        match = re.search(pattern, self.auth_url)
-        self.api_server_url = match.group(0)
-
+    def __init__(self, key):
         # Necessary token
         self.token = self.get_token()
```
Comment (Danmcaires) [inline src/api_request.py:127]: The get_token function uses the auth_url and other variables that were removed from _init_, thus the self.token also needs to be moved.

File: src/api_request.py [hunk 2]
```diff
@@ -37,9 +24,6 @@ def __init__(self, user_query, key, instance):
             secure_oam = self.oam_ip.replace("http://", "https://")
```
Comment (Danmcaires) [inline src/api_request.py:24]: self.oam_ip was removed from __init__

**Review by Danmcaires — CHANGES_REQUESTED** [review]

File: src/api_request.py (hunk 1 above)
Comment (ldeataid) [reply src/api_request.py]: Done.

**Review by Danmcaires — APPROVED** [review]

**Review by borges2157 — APPROVED** [review]
lgtm

**Pull request merged**